### DISCUSSION TITLE
MapClientEntity.getProtocolMappers()

### DISCUSSION THREAD

**Discussion by vramik** [keycloak/keycloak#11135] [Archive]
During review of  https://github.com/keycloak/keycloak/pull/11120 I've noticed that there is a following signature in `MapClientEntity`:

```
Map<String, MapProtocolMapperEntity> getProtocolMappers();
```

Shouldn't it rather be `Set<MapProtocolMapperEntity> getProtocolMappers();` to avoid storing ids twice?

**Comment by hmlnarik** [comment]
I believe it should

**Reply by vramik** [reply]
Thank you, created: https://github.com/keycloak/keycloak/issues/11136

**Comment by martin-kanis** [comment]
Sounds good! Maybe we can even consolidate related `MapProtocolMapperEntity getProtocolMapper(String id)` method so it matches method in `MapClientScopeEntity`.
```
Optional<MapProtocolMapperEntity> getProtocolMapper(String id);
```
